File: introductory_problems/increasing_array/IncreasingArray.java
import java.util.*;

public class IncreasingArray {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = Integer.valueOf(input.nextLine());
        String[] nums = input.nextLine().split(" ");
        input.close();
        findLeastMove(n, nums);
    }

    private static void findLeastMove(int n, String[] nums) {
        long res = 0;
        int curMax = Integer.valueOf(nums[0]);
        for (int i = 1; i < n; i++) {
            int cur = Integer.valueOf(nums[i]);
            if (cur <= curMax) {
                res += (long)(curMax - cur);
            } else {
                curMax = cur;
            }
        }
        System.out.println(res);
    }
}

File: introductory_problems/increasing_array/IncreasingArray.py
n = int(input())
nums = [int(x) for x in input().split()]

cur_max = nums[0]
res = 0
for i in range(1, n):
    if nums[i] <= cur_max:
        res += cur_max - nums[i]
    else:
        cur_max = nums[i]

print(res)